src/main/java/org/generationcp/middleware/dao/GermplasmSearchDAO.java

```diff
@@ -1358,16 +1383,24 @@ private boolean addPreFilteredGids(final GermplasmSearchRequest germplasmSearchR
 			while (iterator.hasNext()) {
 				final Map.Entry<String, String> entry = iterator.next();
 				queryBuilder.append(String.format("a.gid in (select a.gid from atributs a \n"
-					+ " inner join udflds u on a.atype = u.fldno \n"
-					+ " where u.fcode = :attributeKey%s and aval like :attributeValue%<s )", entry.getKey()));
+					+ " INNER JOIN cvterm cv on a.atype = cv.cvterm_id \n"
+					+ " INNER JOIN cvtermprop cp ON cp.type_id = " + TermId.VARIABLE_TYPE.getId() + " and cv.cvterm_id = cp.cvterm_id "
+					+ " LEFT JOIN variable_overrides vo ON vo.cvterm_id = cv.cvterm_id AND vo.program_uuid = :programUUID " //
+					+ " WHERE cp.value in (select name from cvterm where cvterm_id in ("
+					+ VariableType.GERMPLASM_PASSPORT.getId() + ","
+					+ VariableType.GERMPLASM_ATTRIBUTE.getId() + ")) " //
+						+ " AND ( cv.name = :attributeKey%s  or vo.alias = :attributeKey%s ) and a.aval like :attributeValue%<s )",
+					entry.getKey(), entry.getKey()));
 				if (iterator.hasNext()) {
 					queryBuilder.append(" and ");
 				}
 			}
 			queryBuilder.append(LIMIT_CLAUSE);
 
 			final SQLQuery sqlQuery = this.getSession().createSQLQuery(queryBuilder.toString());
+			sqlQuery.setParameter("programUUID", programUUID);
 			for (final Map.Entry<String, String> entry : attributes.entrySet()) {
+				sqlQuery.setParameter("attributeKey" + entry.getKey(), entry.getKey());
```

Comment:
Is this duplicate setting of the parameter` "attributeKey" + entry.getKey()` necessary?

Reply:
done.